datasource/opennms-common/src/main/java/org/opennms/oce/datasource/common/inventory/script/AbstractScriptedInventory.java

```diff
@@ -0,0 +1,199 @@
+
+    private long scriptCacheMillis;
+
+    public AbstractScriptedInventory(String scriptPath, long scriptCacheMillis, BundleContext bundleContext) {
+        if (scriptPath == null) {
+            throw new IllegalArgumentException("Null value for scriptFile.");
+        }
+
+        String script;
+        String scriptExtension;
+
+        this.scriptCacheMillis = scriptCacheMillis;
+
+        if (scriptPath.isEmpty()) {
+            // load default from classpath
+            usingClasspathScript = true;
+            URL scriptUrl = bundleContext.getBundle().getResource(DEFAULT_SCRIPT);
+            StringBuilder sb = new StringBuilder();
+            try (BufferedReader br = new BufferedReader(new InputStreamReader(scriptUrl.openStream()))) {
+                String line;
+                while ((line = br.readLine()) != null) {
+                    sb.append(line).append("\n");
+                }
+            } catch (IOException e) {
+                throw new IllegalArgumentException("Cannot find script in classpath : " + e.getMessage());
+            }
+            script = sb.toString();
+            this.scriptPath = DEFAULT_SCRIPT;
+            scriptExtension = "groovy";
+            LOG.info("Loaded inventory.groovy from the classpath");
+        } else {
+            // read the script from the file system
+            this.scriptPath = scriptPath;
+
+            File file = new File(scriptPath);
+            if (!file.canRead()) {
+                throw new IllegalStateException("Cannot read script at '" + file + "'.");
+            }
+            try {
+                byte[] fileBytes = java.nio.file.Files.readAllBytes(file.toPath());
+                script = new String(fileBytes);
+                scriptExtension = Files.getFileExtension(scriptPath);
+                scriptFileTimestamp = file.lastModified();
+                LOG.info("Loaded script {} from {} with timestamp: {}", file, scriptPath, scriptFileTimestamp);
+            } catch (IOException e) {
+                throw new IllegalStateException("IOError while Reading reading script at '" + file + "'.");
+            }
+            LOG.info("Loaded script {} from the file system", scriptPath);
+        }
+
+        if (bundleContext == null) {
+            manager = new ScriptEngineManager();
+        } else {
+            context = bundleContext;
+            manager = new OSGiScriptEngineManager(context);
+        }
+
+        engine = manager.getEngineByExtension(scriptExtension);
+        if (engine == null) {
+            throw new IllegalStateException("No engine found for extension: " + scriptExtension);
+        }
+
+        try {
+            engine.eval(script);
+        } catch (ScriptException e) {
+            throw new IllegalStateException("Failed to eval() script file - " + this.scriptPath, e);
+        }
+
+        LOG.info("ScriptEngine Initialized with {}", this.scriptPath);
+        invocable = (Invocable) engine;
```

Comment:
We should test that the engine is Invocable before casting and throw an appropriate exception if it is not.

Reply:
done